packages/bot-skeleton/src/scratch/utils/index.js

```diff
@@ -587,6 +587,36 @@ export const isDarkRgbColour = string_rgb => {
 };
 /* eslint-enable */
 
+export const modifyBlockOnCollapse = block_instance => {
+    try {
+        // Return if the block is not collapsed
+        if (!block_instance.collapsed_) return;
+        const [block_image, block_name] = block_instance?.inputList[0]?.fieldRow.map(field => field.value_);
+        const collapsed_field = block_instance?.getField('_TEMP_COLLAPSED_FIELD');
+        const collapsed_input = block_instance?.getInput('_TEMP_COLLAPSED_INPUT');
```

Comment:
Let's use the Constant values from Blockly

Reply:
done ✅